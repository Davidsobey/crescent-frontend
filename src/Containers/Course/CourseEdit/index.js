/**
 *
 * CourseEdit
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import { connect } from 'react-redux';

import Card from '../../../Components/Card';
import TextField from '../../../Components/TextField';
import Button from '../../../Components/Button';
import CourseActions from '../../../Actions/CourseActions';

/* eslint-disable react/prefer-stateless-function */
let CourseEdit = (props) => {
  const submit = (values) => {
    props.dispatch(CourseActions.editCourse(values));
  };
  CourseEdit.propTypes = {
    dispatch: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func,
  };
  return (
    <Card width="600px" title="Edit Course">
      <form onSubmit={props.handleSubmit(submit)} noValidate autoComplete="off">
        <div>
          <div>
            <Field
              name="name"
              label="Course Name"
              margin="normal"
              component={TextField}
            />
          </div>
          <div>
            <Field
              name="description"
              label="Course Description"
              margin="normal"
              component={TextField}
            />
          </div>
          <div>
            <Field
              name="cpdPoints"
              label="Course CPD Points"
              margin="normal"
              component={TextField}
            />
          </div>
        </div>
        <div className="alignRight">
          <Button variant="raised" color="primary" type="submit">
            Edit Course
          </Button>
        </div>
      </form>
    </Card>
  );
};

const mapDispatchToProps = dispatch => ({
  dispatch,
});

// Decorate with reduxForm(). It will read the initialValues prop provided by connect()
CourseEdit = reduxForm({
  form: 'courseEdit', // a unique identifier for this form
})(CourseEdit);

// You have to connect() to any reducers that you wish to connect to yourself
const FormState = connect(
  state => ({
    initialValues: state.CourseReducer.course, // pull initial values from reducer
  }),
  mapDispatchToProps,
)(CourseEdit);

export default FormState;
