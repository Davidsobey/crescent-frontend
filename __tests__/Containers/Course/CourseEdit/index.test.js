import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import CourseEdit from '../../../../src/Containers/Course/CourseEdit/index.js';


const mockStore = configureStore();
const initialState = {
  CourseReducer:{
    course: null,
    course_loading: false,
    course_editing: false,
  }
};
const store = mockStore(initialState);

describe('<CourseEdit />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<CourseEdit store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});