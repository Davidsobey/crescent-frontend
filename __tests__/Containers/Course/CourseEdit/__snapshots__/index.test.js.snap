// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseEdit /> render() renders the component 1`] = `
<Connect(Form(CourseEdit))
  course_editing={false}
  course_loading={false}
  destroyOnUnmount={true}
  dispatch={[Function]}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="courseEdit"
  getFormState={[Function]}
  initialValues={null}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
